python/lsst/meas/astrom/pessimistic_pattern_matcher_b_3D.py

```diff
@@ -1008,17 +1033,17 @@ def _test_spoke(self, src_ctr, src_delta, src_dist, src_ctr_delta,
             # its direction. To do that we need to know the sine as well.
             # This cross product calculation does that.
             cross_ref = ref_sign * (
-                np.cross(ref_delta_array[ref_dist_idx], ref_delta) /
-                (ref_dist_array[ref_dist_idx] * ref_dist))
-            dot_cross_ref = np.dot(cross_ref, ref_ctr)
+                np.cross(proj_ref_delta, proj_ref_ctr_delta) /
+                ref_geom_dist)
+            sin_theta_ref = np.dot(cross_ref, ref_ctr)
 
             # Check the value of the cos again to make sure that it is not
             # near zero.
             if abs(cos_theta_src) < src_sin_tol:
-                sin_comparison = (dot_cross_src - dot_cross_ref) / src_sin_tol
+                sin_comparison = (sin_theta_src - sin_theta_ref) / src_sin_tol
             else:
                 sin_comparison = \
-                    (dot_cross_src - dot_cross_ref) / cos_theta_ref
+                    (sin_theta_src - sin_theta_ref) / cos_theta_ref
 
             if abs(sin_comparison) > src_sin_tol:
                 continue
```

Comment:
Does the unit test not require any changes?
Please run it through Jenkins either way. I would suggest testing ci_hsc if it's used in it at all.
Consider filing a tech debt ticket to make the rest of the file more consistent with the style guide.

Reply:
All the unittests current pass for each of  commits in this ticket and Jenkins will be run before this ticket is merged. The line you comment on here is purely a cosmetic change as before doing the plane project the values of the sines were approximate. Now that they are exact, I've changed the name to explicitly say sine.

As for ci_hsc, the code config uses another matcher in ci_hsc. I'm working on another ticket, DM-14857, to make the pessimisticMatcher the default.